flink-1.17-connector-bigquery/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/RowDataToProtoSerializerTest.java

```diff
@@ -0,0 +1,517 @@
+     *   <li>"FLOAT" - FLOAT
+     *   <li>"STRING" - STRING
+     *   <li>"BOOLEAN" - BOOLEAN
+     *   <li>"BYTES" - BYTES
+     * </ul>
+     */
+    @Test
+    public void testAllBigQuerySupportedPrimitiveTypesConversionToDynamicMessageCorrectly()
+            throws BigQuerySerializationException {
+        // Obtaining the Schema Provider and the Row Data Record.
+        BigQuerySchemaProvider bigQuerySchemaProvider =
+                TestBigQuerySchemas.getSchemaWithRequiredPrimitiveTypes();
+        Schema avroSchema = bigQuerySchemaProvider.getAvroSchema();
+        Descriptors.Descriptor descriptor = bigQuerySchemaProvider.getDescriptor();
+        LogicalType logicalType =
+                BigQuerySchemaProviderTableImpl.getDataTypeSchemaFromAvroSchema(avroSchema)
+                        .getLogicalType();
+        RowDataToProtoSerializer rowDataSerializer = new RowDataToProtoSerializer();
+        rowDataSerializer.init(bigQuerySchemaProvider);
+        rowDataSerializer.setLogicalType(logicalType);
+
+        byte[] byteArray = "Any String you want".getBytes();
+        GenericRowData row = new GenericRowData(6);
+        row.setField(0, -7099548873856657385L);
+        row.setField(1, 0.5616495161359795);
+        row.setField(2, StringData.fromString("String"));
+        row.setField(3, true);
+        row.setField(4, byteArray);
+        GenericRowData innerRow = new GenericRowData(1);
+        innerRow.setField(0, StringData.fromString("hello"));
+        row.setField(5, innerRow);
+
+        // Form the Dynamic Message.
+        DynamicMessage message =
+                rowDataSerializer.getDynamicMessageFromRowData(row, descriptor, logicalType);
+
+        // Check for the desired results.
+        assertEquals(-7099548873856657385L, message.getField(descriptor.findFieldByNumber(1)));
+        assertEquals(0.5616495161359795, message.getField(descriptor.findFieldByNumber(2)));
+        assertEquals("String", message.getField(descriptor.findFieldByNumber(3)));
+        assertEquals(true, message.getField(descriptor.findFieldByNumber(4)));
+        assertEquals(
+                ByteString.copyFrom(byteArray), message.getField(descriptor.findFieldByNumber(5)));
+        Descriptors.FieldDescriptor fieldDescriptor = descriptor.findFieldByNumber(6);
+        message = (DynamicMessage) message.getField(fieldDescriptor);
+        assertEquals(
+                "hello",
+                message.getField(
+                        descriptor
+                                .findNestedTypeByName(fieldDescriptor.toProto().getTypeName())
+                                .findFieldByNumber(1)));
+    }
+
+    /**
+     * Test to check <code>getDynamicMessageFromRowData()</code> for Primitive types supported by
+     * BigQuery. However <code>null</code> value is passed to the record field to test for error.
+     */
+    @Test
+    public void testAllBigQuerySupportedPrimitiveTypesConversionToDynamicMessageIncorrectly() {
```

Comment:
can we also add a comparable test for nullable fields where we expect conversion to no-op?

Reply:
jk i see you have it further down :)